iroh-relay/src/protos/relay.rs

```diff
@@ -411,108 +205,55 @@ impl Frame {
         + payload_len
     }
 
+    /// Tries to decode a frame received over websockets.
+    ///
+    /// Specifically, bytes received from a binary websocket message frame.
     #[allow(clippy::result_large_err)]
-    pub(crate) fn from_bytes(bytes: Bytes, cache: &KeyCache) -> Result<Self, RecvError> {
-        if bytes.is_empty() {
-            return Err(TooSmallSnafu.build());
-        }
-        let frame_type = FrameType::from(bytes[0]);
-        let content = bytes.slice(1..);
-        let res = match frame_type {
-            FrameType::ClientInfo => {
-                if content.len() < PublicKey::LENGTH + Signature::BYTE_SIZE + MAGIC.len() {
-                    return Err(InvalidFrameSnafu.build());
-                }
-                if &content[..MAGIC.len()] != MAGIC.as_bytes() {
-                    return Err(InvalidFrameSnafu.build());
-                }
+    pub(crate) fn from_bytes(mut content: Bytes, cache: &KeyCache) -> Result<Self, Error> {
+        let frame_type = FrameType::from_bytes(&mut content)?;
+        let frame_len = content.len();
+        snafu::ensure!(
+            frame_len <= MAX_PACKET_SIZE,
+            FrameTooLargeSnafu { frame_len }
+        );
 
-                let start = MAGIC.len();
-                let client_public_key =
-                    cache.key_from_slice(&content[start..start + PublicKey::LENGTH])?;
-                let start = start + PublicKey::LENGTH;
-                let signature =
-                    Signature::from_slice(&content[start..start + Signature::BYTE_SIZE])?;
-                let start = start + Signature::BYTE_SIZE;
-                let message = content.slice(start..);
-                Self::ClientInfo {
-                    client_public_key,
-                    message,
-                    signature,
-                }
-            }
-            FrameType::SendPacket => {
-                if content.len() < PublicKey::LENGTH {
-                    return Err(InvalidFrameSnafu.build());
-                }
-                let frame_len = content.len() - PublicKey::LENGTH;
-                if frame_len > MAX_PACKET_SIZE {
-                    return Err(FrameTooLargeSnafu { frame_len }.build());
-                }
-
-                let dst_key = cache.key_from_slice(&content[..PublicKey::LENGTH])?;
-                let packet = content.slice(PublicKey::LENGTH..);
-                Self::SendPacket { dst_key, packet }
-            }
+        let res = match frame_type {
             FrameType::RecvPacket => {
-                if content.len() < PublicKey::LENGTH {
-                    return Err(InvalidFrameSnafu.build());
-                }
+                snafu::ensure!(content.len() >= NodeId::LENGTH, InvalidFrameSnafu);
```

Comment:
I think you also want to be careful to reject empty payloads, because that would allow sending 0-sized datagrams to other clients and I have no idea how all things would cope with that.  I'm assuming on the internet this isn't allowed.

So maybe change this to `>`?

Reply:
Hmm. Yeah. I went back and forth on this, but I guess as long as there's no need for empty payloads, I shouldn't have them.